src/inference/src/threading/ie_cpu_streams_executor.cpp

```diff
@@ -78,103 +78,170 @@ struct CPUStreamsExecutor::Impl {
-                    // Special handling of _threads_per_stream_small == 3
-                    const auto small_core_id = small_core_skip ? 0 : streamId_wrapped - big_core_streams;
-                    const auto stream_id =
-                        hybrid_core
-                            ? (small_core ? small_core_id
-                                          : (logic_core ? streamId_wrapped - phy_core_streams : streamId_wrapped))
-                            : streamId_wrapped;
-                    const auto thread_binding_step = hybrid_core ? (small_core ? _impl->_config._threadBindingStep : 2)
-                                                                 : _impl->_config._threadBindingStep;
-                    // Special handling of _threads_per_stream_small == 3, need to skip 4 (Four cores share one L2 cache
-                    // on the small core), stream_id = 0, cpu_idx_offset cumulative plus 4
-                    const auto small_core_offset =
-                        small_core_skip ? _impl->_config._small_core_offset + (streamId_wrapped - big_core_streams) * 4
-                                        : _impl->_config._small_core_offset;
-                    const auto cpu_idx_offset =
-                        hybrid_core
-                            // Prevent conflicts with system scheduling, so default cpu id on big core starts from 1
-                            ? (small_core ? small_core_offset : (logic_core ? 0 : 1))
-                            : 0;
-
-                    _taskArena.reset(new custom::task_arena{custom::task_arena::constraints{}
-                                                                .set_core_type(selected_core_type)
-                                                                .set_max_concurrency(max_concurrency)});
+                }
+                if (_impl->_config._bind_cores) {
+                    const auto cpu_core_type =
+                        _streamId < _impl->_config._big_core_streams
+                            ? MAIN_CORE_PROC
+                            : (_streamId < _impl->_config._big_core_streams + _impl->_config._big_core_logic_streams
+                                   ? HYPER_THREADING_PROC
+                                   : EFFICIENT_CORE_PROC);
+                    const auto small_core_threads_3 = cpu_core_type == EFFICIENT_CORE_PROC && concurrency == 3 &&
+                                                      _impl->_config._small_core_streams > 1;
+                    const auto thread_binding_step = getThreadStep(cpu_core_type);
+                    const auto cpu_idx_offset = getCoreOffset(cpu_core_type);
+                    for (int i = 0; i < concurrency; i++) {
+                        _cpu_ids.push_back(cpu_idx_offset + thread_binding_step * i);
+                    }
+                    if (small_core_threads_3) {
+                        _cpu_ids.push_back(cpu_idx_offset + 3);
+                    }
```

Comment:
Please search available processor purely by look-up table instead of calculation based on "offset + step * n"

Reply:
Yes, I changed the method of getting "_cpu_ids".